### PR TITLE
sbt resolves dependencies every compile when using %% with dependencyOverrides

### PR DESCRIPTION
**Pull request by eed3si9n**
The issue is not caused when % is used instead of %%.

Following is reproducible build.sbt.
https://gist.github.com/tkawachi/125cdcad0b0e014f1f44

ref. https://groups.google.com/d/msg/sbt-dev/cUOCbLVXwEQ/O9lzi0atI_kJ

### CODE REVIEW

**Comment by eed3si9n** [comment]
Attached PR on this. /review @jsuereth

File: ivy/src/main/scala/sbt/IvyConfigurations.scala [hunk 1]
```diff
@@ -108,18 +138,18 @@ object InlineConfiguration {
...
+    val configurations: Seq[Configuration],
+    val defaultConfiguration: Option[Configuration],
+    val ivyScala: Option[IvyScala],
+    val validate: Boolean,
+    val conflictManager: ConflictManager) extends ModuleSettings {
   def withConfigurations(configurations: Seq[Configuration]) = copy(configurations = configurations)
   def noScala = copy(ivyScala = None)
 
-  def copy(module: ModuleID = this.module,
+  private[sbt] def copy(module: ModuleID = this.module,
```
Comment (jsuereth) [inline ivy/src/main/scala/sbt/IvyConfigurations.scala:152]: nice Binary compat trick here :)

**Comment by jsuereth** [comment]
LGTM

**Pull request merged**